app/email.go

```diff
@@ -373,26 +374,32 @@ func (es *EmailService) SendInviteEmails(team *model.Team, senderName string, se
 
 			if err := es.srv.Store.Token().Save(token); err != nil {
 				mlog.Error("Failed to send invite email successfully ", mlog.Err(err))
+				errorMails = append(errorMails, invite)
 				continue
 			}
 			bodyPage.Props["Link"] = fmt.Sprintf("%s/signup_user_complete/?d=%s&t=%s", siteURL, url.QueryEscape(data), url.QueryEscape(token.Token))
 
 			if err := es.sendMail(invite, subject, bodyPage.Render()); err != nil {
 				mlog.Error("Failed to send invite email successfully ", mlog.Err(err))
+				errorMails = append(errorMails, invite)
 			}
 		}
 	}
+	if len(errorMails) > 0 {
+		return model.NewAppError("emailSentFailed", "api.user.invite.error", nil, strings.Join(errorMails, ", "), http.StatusUnprocessableEntity)
```

Comment:
I think `422` is a little bit misleading here. We can parse errors of `es.sendMail` or we can pick the shortest path:

```suggestion
		return model.NewAppError("emailSentFailed", "api.user.invite.error", nil, strings.Join(errorMails, ", "), http.StatusInternalServerError)
```

Reply:
i chose [unprocessable entity](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/422) as some might fail but not all, so I didn't want to imply that all has go wrong or something failed on the server, it is just that something is not right with some section of the request. 

Anyways I'm 0/5 on this and I already expected it might be preferred to have a 500 error, so i'll change that.